src/services/addressResolver/index.ts

```diff
@@ -43,8 +43,8 @@ export const getIdentityName = async (addresses: ThirdPartyAPIEntryProps[], addr
       currentValue.apiHeader,
       currentValue.apiKey
     );
-    return result;
-  }, Promise.resolve(undefined));
+    return { result, source: currentValue.name };
+  }, Promise.resolve<{ result: string; source: string } | undefined>(undefined));
```

Comment:
{ result: string, source: string } is a type that's common to both types of resolvers, do you think it makes sense to  extract this type and reuse it so that people don't make mistakes down the road?

Reply:
yes boss